packages/contracts-bedrock/contracts/universal/StandardBridge.sol

```diff
@@ -344,18 +362,34 @@ abstract contract StandardBridge {
         bytes calldata _extraData
     ) public onlyOtherBridge returns (bool) {
         try this.completeOutboundTransfer(_localToken, _remoteToken, _to, _amount) {
-            emit ERC20BridgeFinalized(_localToken, _remoteToken, _from, _to, _amount, _extraData);
-            return true;
+            if (_from == address(otherBridge)) {
+                // The _from address is the address of the remote bridge if a transfer fails to be
+                // finalized on the remote chain.
+                // slither-disable-next-line reentrancy-events
+                emit ERC20BridgeRefunded(_localToken, _remoteToken, _to, _amount, _extraData);
+                return true;
+            } else {
+                emit ERC20BridgeFinalized(
+                    _localToken,
+                    _remoteToken,
+                    _from,
+                    _to,
+                    _amount,
+                    _extraData
+                );
+                return true;
+            }
         } catch {
             // Something went wrong during the bridging process, return to sender.
             // Can happen if a bridge UI specifies the wrong L2 token.
-            // We reverse the to and from addresses to make sure the tokens are returned to the
-            // sender on the other chain and preserve the accuracy of accounting based on emitted
+            // Since the _to address never received the L2 token, we set this bridge contract
```

Comment:
I'm a little confused by this comment. The original flow makes the accounting look like:

(person A bridged X amount of token Y to person B from chain 1 to chain 2) 
(person B bridged X amount of token Y to person A from chain 2 to chain 1)

Which seems very accurate. I guess the difference here is the explicit acknowledgement that the user did NOT actually transfer the funds back and instead there was a refund because of an error on the other chain.

Reply:
Correct. It's not as thought Person B actually sent anything.